ldes-server-domain/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/domain/ldesfragment/services/FragmentationQueueMediatorImpl.java

```diff
@@ -4,36 +4,34 @@
 import org.springframework.scheduling.annotation.Scheduled;
 import org.springframework.stereotype.Component;
 
-import java.util.concurrent.ExecutorService;
-import java.util.concurrent.Executors;
 import java.util.concurrent.LinkedBlockingQueue;
 import java.util.concurrent.atomic.AtomicInteger;
 
-@Component
-public class FragmentationQueueMediatorImpl implements FragmentationQueueMediator {
-
-	private final ExecutorService executorService = Executors.newSingleThreadExecutor();
-	private final LinkedBlockingQueue<String> ldesMembersToFragment = new LinkedBlockingQueue<>();
```

Comment:
Would it not have been easier/cleaner to inject an interface `LdesFragmentQueue` here?
That interface could have a limited set of operations (the ones that are used here):
* add(String memberId)
* poll()

All code would remain the same but you could plug in RabbitMQ by providing an implementation of `LdesFragmentQueue`: `LdesFragmentRabbitMQ implements LdesFragmentQueue`

Reply:
As far as i've understood how rabbitmq listener, this is not a poll based mechanic. And by treating those modules just as a queue, you will make them dumber than needs be in my opinion. 
This also allows for the most modularity (e.g. poll vs push based) for minimal code to write yourself